openeo_processes_dask/process_implementations/cubes/load.py

```diff
@@ -0,0 +1,164 @@
+import datetime
+import json
+from collections.abc import Iterator
+from pathlib import PurePosixPath
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from urllib.parse import unquote, urljoin, urlparse
+
+import planetary_computer as pc
+import pyproj
+import pystac_client
+import stackstac
+import xarray as xr
+from openeo_pg_parser_networkx.pg_schema import *
+from stac_validator import stac_validator
+
+from openeo_processes_dask.process_implementations.cubes._filter import (
+    _reproject_bbox,
+    filter_bands,
+    filter_bbox,
+    filter_temporal,
+)
+from openeo_processes_dask.process_implementations.data_model import RasterCube
+from openeo_processes_dask.process_implementations.exceptions import (
+    NoDataAvailable,
+    TemporalExtentEmpty,
+)
+
+# "NoDataAvailable": {
+#     "message": "There is no data available for the given extents."
+# },
+# "TemporalExtentEmpty": {
+#     "message": "The temporal extent is empty. The second instant in time must always be greater/later than the first instant in time."
+# }
+__all__ = ["load_stac"]
+
+
+def load_stac(
+    url: str,
+    spatial_extent: BoundingBox = None,
+    temporal_extent: Optional[
+        list[Union[str, datetime.datetime, datetime.date]]
+    ] = None,
+    bands: Optional[list[str]] = None,
+    properties: Optional[dict] = None,
+    **kwargs,
+) -> RasterCube:
+    def validate_stac(url):
+        stac = stac_validator.StacValidate(url)
+        is_valid_stac = stac.run()
+        if not is_valid_stac:
+            raise Exception(
+                f"The provided link is not a valid STAC. stac-validator message: {stac.message}"
+            )
+        if len(stac.message) == 1:
+            try:
+                asset_type = stac.message[0]["asset_type"]
+            except:
+                raise Exception(f"stac-validator returned an error: {stac.message}")
+        else:
+            raise Exception(
+                f"stac-validator returned multiple items, not supported yet. {stac.message}"
+            )
+        return is_valid_stac, asset_type
```

Comment:
why return `is_valid_stac`? it will always be true, because an exception is thrown above if it is false